lib/model/query/entities.js

```diff
@@ -42,28 +45,39 @@ const nextval = sql`nextval(pg_get_serial_sequence('entities', 'id'))`;
 // standard authenticated API request. The worker has better access to the event
 // actor/initiator and actee/target so it will do the logging itself (including
 // error logging).
-const createNew = (partial, subDef) => ({ one }) => {
+const createNew = (partial, subDef, userAgentIn) => ({ one, context }) => {
+  let creatorId;
+  let userAgent;
+  let subDefId;
+
+  // Set creatorId and userAgent from submission def if it exists
+  if (subDef != null && subDef.id != null)
+    ({ id: subDefId, submitterId: creatorId, userAgent } = subDef);
+  else {
+    creatorId = context.auth.actor.map((actor) => actor.id).orNull();
+    userAgent = blankStringToNull(userAgentIn);
+    subDefId = null;
+  }
+
   const json = JSON.stringify(partial.def.data);
+
   return one(sql`
-with def as (${_defInsert(nextval, subDef.id, subDef.submitterId, subDef.userAgent, partial.def.label, json)}),
-ds as (${_getDataset(partial.aux.dataset, subDef.id)}),
+with def as (${_defInsert(nextval, subDefId, creatorId, userAgent, partial.def.label, json)}),
+ds as (${_getDataset(partial.aux.dataset, subDefId, partial.datasetId)}),
 ins as (insert into entities (id, "datasetId", "uuid", "createdAt", "creatorId")
-  select def."entityId", ds."id", ${partial.uuid}, def."createdAt", ${subDef.submitterId} from def
-  join ds on ds."subDefId" = def."submissionDefId"
+  select def."entityId", ds."id", ${partial.uuid}, def."createdAt", ${creatorId} from def, ds
   returning entities.*)
 select ins.*, def.id as "entityDefId", ds."acteeId" as "dsActeeId", ds."name" as "dsName" from ins, def, ds;`)
-    .then(({ entityDefId, dsActeeId, dsName, ...entityData }) => // TODO/HACK: reassembly inspired by Submissions.createNew
+    .then(({ entityDefId, dsActeeId, dsName, ...entityData }) => // TODO/HACK: reassemble just enough to log audit event
       new Entity(entityData, {
-        def: new Entity.Def({
+        currentVersion: new Entity.Def({
           id: entityDefId,
           entityId: entityData.id,
-          submissionDefId: subDef.id,
-          createdAt: entityData.createdAt,
-          creatorId: subDef.submitterId,
+          submissionDefId: subDefId,
           data: partial.def.data,
           label: partial.def.label
         }),
-        dataset: { acteeId: dsActeeId, name: dsName }
+        dataset: new Dataset({ acteeId: dsActeeId, name: dsName })
```

Comment:
I don't quite know what to return here. Either
- just enough for audit logging
OR
- EVERYTHING the API response will need

Currently went with enough details for audit logging.